src/components/categories/Categories.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import './categories.css';
+const Categories = ({ filterItems, library }) => {
+  // The purpose of this three lines is to convert the tags
+  // that are in subarray into one array, so that i can map through it
+  const listOfTags = library.map((item) => item.tags);
+  const listOfTagsIntoOneString = listOfTags.toString();
+  const convertedListOfTags = listOfTagsIntoOneString.split(',');
+
+  const allCategories = [
+    'Tous',
+    ...new Set(convertedListOfTags.map((item) => item)),
+  ];
```

Comment:
I'm not fan of adding comments. It means that code is not readable. Here, I think they are not needed

Same as above, you can do it one line. I'd suggest not adding intermediate variables only when needed for readability. Otherwise, you are using memory which is not needed.

```suggestion
  const allCategories = [
    'Tous',
    ...new Set(
      library
        .map((item) => item.tags)
        .toString()
        .split(',')
        .map((item) => item)
    ),
  ];
```